experiment/manual-trigger/manual-trigger.go

```diff
@@ -0,0 +1,178 @@
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+type options struct {
+	githubEndpoint     string
+	githubToken        string
+	jenkinsBearerToken string
+	jenkinsEndpoint    string
+	jobName            string
+	num                int
+	org                string
+	repo               string
+}
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.githubEndpoint, "githubapiurl", "https://api.github.com", "GitHub's API endpoint.")
+	flag.StringVar(&o.jenkinsEndpoint, "jenkinsurl", "", "Jenkins url.")
+
+	flag.StringVar(&o.githubToken, "ghtoken", "", "Path to file containing GitHub token.")
+	flag.StringVar(&o.jenkinsBearerToken, "jbtoken", "", "Path to file containing Jenkins Bearer token.")
+
+	flag.StringVar(&o.jobName, "jobname", "", "Name of Jenkins job")
+
+	flag.IntVar(&o.num, "num", 0, "GitHub issue number")
+	flag.StringVar(&o.org, "org", "", "GitHub organization")
+	flag.StringVar(&o.repo, "repo", "", "GitHub repository")
+	flag.Parse()
+	return o
+}
+
+func sanityCheckFlags(o options) error {
+	if o.num <= 0 {
+		return fmt.Errorf("empty or invalid --num")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --org")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --repo")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("empty --ghtoken")
+	}
+	if o.jenkinsBearerToken == "" {
+		return fmt.Errorf("empty --jbtoken")
+	}
+	if o.jenkinsBearerToken == "" {
+		return fmt.Errorf("empty --jobname")
+	}
+
+	_, err := url.Parse(o.githubEndpoint)
+	if o.githubEndpoint == "" || err != nil {
+		return fmt.Errorf("must specify a valid URL to --githubApiUrl")
+	}
+
+	_, err = url.Parse(o.jenkinsEndpoint)
+	if o.jenkinsEndpoint == "" || err != nil {
+		return fmt.Errorf("must specify a valid URL to --jenkinsurl")
+	}
+
+	return nil
+}
+
+func readToken(path string) (string, error) {
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(contents)), nil
+}
+
+func main() {
+	o := flagOptions()
+	err := sanityCheckFlags(o)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	jbtoken, err := readToken(o.jenkinsBearerToken)
+	if err != nil {
+		log.Fatalf("cannot read file specified by --jbtoken %v", err)
+	}
+
+	ghtoken, err := readToken(o.githubToken)
+	if err != nil {
+		log.Fatalf("cannot read file specified by --ghtoken: %v", err)
+	}
+
+	auth := jenkins.BearerTokenAuthConfig{
+		Token: jbtoken,
+	}
+
+	authconfig := jenkins.AuthConfig{
+		Basic:       nil,
+		BearerToken: &auth,
+	}
+
+	jc := jenkins.NewClient(o.jenkinsEndpoint, &authconfig)
+
+	gc := github.NewClient(ghtoken, o.githubEndpoint)
+
+	fmt.Println("Querying GitHub for details...")
+	pr, err := gc.GetPullRequest(o.org, o.repo, o.num)
+	if err != nil {
+		log.Fatalf("Unable to get specified pull request %v", err)
+	}
+
+	ref, err := gc.GetRef(o.org, o.repo, "heads/"+pr.Base.Ref)
+	if err != nil {
+		log.Fatalf("Unable to get reference to pull request %v", err)
+	}
+	fmt.Println("done")
```

Comment:
These `Print` statements are what i meant before -- if it's critical information for run-time debugging, use a logger and log them. But if it's just a comment like `"done"` I don't think we should keep them in the final version of this tool